src/utils/sound_loader.py

```diff
@@ -1,7 +1,21 @@
 from src.conf import config
-
+import math
+import numpy as np
+import scipy.signal as sig
 
 class SoundLoader:
     def __init__(self):
         # tak się dobiera do ustalonych parametrów (wszystkie są w configurator.py)
-        print(config.analysis['sampling_frequency'])
\ No newline at end of file
+        print(config.analysis['sampling_frequency'])
+
+    def downsample(self, audio, fs):
+        sampling_frequency = config.analysis['sampling_frequency']
+        nww = (fs * sampling_frequency) / math.gcd(fs, sampling_frequency)
+        up = int(nww // fs)
+        down = int(nww // sampling_frequency)
+        audio_up = np.zeros(len(audio) * up)  # zwiększamy sygnał
+        audio_up[1::up] = audio
```

Comment:
W sumie dlaczego tutaj lecimy od 1?
Wydaje mi się, że lepiej jest lecieć od up // 2.
Wtedy ten nasz sygnał rozszerzony będzie miał bardziej równo rozsiane próbki. Jak jedziemy od 1, to cały sygnał będzie przesunięty bardziej na lewo.